### DISCUSSION TITLE
Add a <link preload rel="image"> for each "critical" image (when using gatsby-image)

### DISCUSSION THREAD

**Discussion by KyleAMathews** [gatsbyjs/gatsby#29877] [Ideas / Feature Requests]
@baobabKoodaa [noticed that critical images can be blocked on slower connections](https://github.com/gatsbyjs/gatsby/issues/12047) by loading JavaScript. We should prioritize loading critical images (and fonts) over loading JS as the above-the-fold content needs to be loaded before we worry about making it interactive.

One way to do that is when SSRing `gatsby-image` capture this and add some code to the `<head>` which can trigger the preloading.

The tricky part here is that the browser doesn't natively handle preloading/prefetching images that use `srcset` like gatsby-image does.

So we'd need to embed some code (as outlined in my RFC https://github.com/gatsbyjs/rfcs/pull/35) that would take a srcset and pick the right image based on the browser screen size / resolution to preload.

**Comment by KyleAMathews** [comment]
Oh https://twitter.com/domfarolino/status/1129591317410045954

https://www.chromestatus.com/feature/5164259990306816

Ok, that makes this a lot easier :-D

**Comment by lesagi** [comment]
Hi,
I've never contributed to open-source before, I'm eager to solve this issue - is it a To-Do task?
Is there someone to guide me through it?

**Comment by quansenB** [comment]
I would also like this being solved and would contribute. However I have no idea where in the code to look, so any hint would be great :)
